Quad grid update

paciorek opened the pull request:
@paul-vdb I think it's safest if we merge in `quadGridUpdate` before merging in `update_naming`, as I assume that `quadGridUpdate` is based on devel before the naming changes.

Are you at a point where it makes sense to merge `quadGridUpdate` in?

paul-vdb committed:
major updates to quadrature grid

Overhaul of quadrature grid code and associated tests. In line with logic from mvQuad and tested for users providing their own quadrature rules. Sparse rule has been tested as well as pruning and made available in approxposterior. Not added officially to Laplace as this has minimal AGHQ options.

Laplace and Approx posterior were tweaked to accommodate this change. Additionally, I added the option of using an identity transformation (no transformation). It may be that some quadrules are not adaptive. This is incomplete in Laplace and needs more work (check weights etc).

In runNestedApprox, I updated `improveMarginals` to include the option to switch from AGHQ to a SPARSE grid or to prune.

paul-vdb committed:
configureQuadGrid control

default to NULL for quadRules as per Chris' comment.

paul-vdb committed:
build for install

Changes to make it install properly and other minor tweaks. An additional check to make sure indices don't go haywire for quadgrids.

paul-vdb committed:
Added prune control element, skew warning

paul-vdb committed:
Note for sparse grid computation

paul-vdb committed:
Fixed dmarginal for jacobian

paul-vdb committed:
better naming

paul-vdb committed:
Updates to runNestedApprox

Added a marginal plot, setParamGrid, and changed input for improveMarginals to control more options and ensure that no changes are made to default inputs.

paul-vdb committed:
Error trapping

Trying to find a bug I think I introduced with my quadrature changes. Crashes the session so probably indexing. Does get triggered with calling the integration free method as first point via runNestedApprox

paul-vdb commented:
@paciorek I agree. I should be ready by end of day.

paciorek commented:
Great, thanks. I'm about to get back to jamming on this, tomorrow and this weekend, so that is good timing.

paul-vdb commented:
@paciorek It will be tomorrow morning. I just resolved a bug in the update to the quad grids, but trying to tackle what is happening with Laplace now just in case it is related.

Commit:
Errors to resolve

Fixed vectorization of modeI in quadGrid
Added check for NA starting values in Laplace
Still failing when trying to run optimize in Laplace for Salamanders.
Indexing error when changing size of hyperGrid. Need to add some more index error checks.

paul-vdb committed:
Adding robustness

Things are crashing in Salamanders example but that has been highlighted as an issue with Laplace and not this update. In theory ready to merge. Runs perfect for existing examples and new quadrature tests.

paul-vdb committed:
Merge branch 'devel' into quadGridUpdate
